### PR TITLE
Show precise number tooltip of leaderboard score rank when 1000 or higher

### PR DESCRIPTION
**Pull request by Joehuu**
Seems to be a good UX improvement.

image

### CODE REVIEW

**Comment by peppy** [comment]
Should probably be formatted to "#16,483" (see other examples of formatting to ensure you support local culture).

File: osu.Game/Online/Leaderboards/LeaderboardScore.cs [hunk 1]
```diff
@@ -356,6 +346,24 @@ public ScoreComponentLabel(LeaderboardScoreStatistic statistic)
             }
         }
 
+        private class RankLabel : Container, IHasTooltip
+        {
+            public RankLabel(int? rank)
+            {
+                TooltipText = rank == null || rank < 1000 ? null : $"#{rank:N0}";
```
Comment (peppy) [inline osu.Game/Online/Leaderboards/LeaderboardScore.cs:353]:
```suggestion
                if (rank > 1000)
                    TooltipText = $"#{rank:N0}";
```

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Online/Leaderboards/LeaderboardScore.cs (hunk 1 above)
Comment (Joehuu) [reply osu.Game/Online/Leaderboards/LeaderboardScore.cs]: I've edited your suggestion as we also need 1000 to be included. `ToMetric` will format numbers 1000-1050 to `1k`.

**Review by peppy — APPROVED** [review]
seems ok

**Pull request merged**